storage/src/main/java/org/apache/kafka/storage/internals/log/LoadedLogOffsets.java

```diff
@@ -18,11 +18,7 @@
 
 import java.util.Objects;
 
-public class LoadedLogOffsets {
-    public final long logStartOffset;
-    public final long recoveryPoint;
-    public final LogOffsetMetadata nextOffsetMetadata;
-
+public record LoadedLogOffsets(long logStartOffset, long recoveryPoint, LogOffsetMetadata nextOffsetMetadata) {
     public LoadedLogOffsets(final long logStartOffset,
                             final long recoveryPoint,
                             final LogOffsetMetadata nextOffsetMetadata) {
```

Comment:
Should we remain `equals()`, `hashCode()`, `toString()` for this record?

Reply:
Removed `equals()` and `hashCode()`